File: examples/lmm_dyestuff.jl
using CSV
using DataFrames
using MixedModels

function main()
    file_path = joinpath("tests", "data", "dyestuff.csv")
    if !isfile(file_path)
        println("Could not find the dataset at: $file_path")
        exit(1)
    end

    println("Loading data from $file_path...")
    df = CSV.read(file_path, DataFrame)

    println("\nFitting model: Yield ~ 1 + (1 | Batch)")
    form = @formula(Yield ~ 1 + (1 | Batch))
    m1 = fit(MixedModel, form, df, REML=true)
    
    println("\n=== Model Summary ===")
    println(m1)
end

main()

File: examples/lmm_dyestuff.py
import polars as pl
import lme_python as lme

def main():
    print("Loading data from tests/data/dyestuff.csv...")
    try:
        df = pl.read_csv("tests/data/dyestuff.csv")
    except FileNotFoundError:
        print("Could not find the dataset. Please run from the root of lme-rs.")
        return

    print("\nFitting model: Yield ~ 1 + (1 | Batch)")
    
    # Fit the Linear Mixed Model
    model = lme.lmer("Yield ~ 1 + (1 | Batch)", df, True)
    
    print("\n=== Model Summary ===")
    print(model.summary())

if __name__ == "__main__":
    main()
